clang-tools-extra/clang-tidy/add_new_check.py

```diff
@@ -754,7 +765,7 @@ def main():
         language_restrict = (
             f"%(lang)s.{cpp_language_to_requirements.get(args.standard, 'CPlusPlus')}"
         )
-    elif language in ["objc", "objc++"]:
+    else:  # "objc" or "objc++"
         language_restrict = "%(lang)s.ObjC"
```

Comment:
This change would result in any language string that is not supported setting the language restriction to `ObjC` (e.g., `--language CUDA`).
I'd say keep the `elif` and do a
```c++
if language_restrict is None:
    print(f"Unsupported language '{language}' was specified")
```
afterward

Reply:
Other languages technically can not get to this point because of `choices = language_to_extension.keys()` which only contains `["c", "c++", "objc", "objc++"]`, but I agree keeping the condition is more robust in case this lists change.